### PR TITLE
Update manage sessions screen

### LINKED ISSUE #1295: Manage Session Settings / Cross Signing update
This issue describes what the manage session list screen should display, and available actions.

This screen depends on whether cross signing is enable in the account, and if the current session is verified

### And Current session is trusted

    - Current Session:
    	- Green Shield, on tap the bottomsheet shows:
    		- Verified
    		- Shows device name / id 
    		- (action) Rename 
    - Other Sessions:
    	- Shields reflects the cross signed status (local trust ignored), on tap the bottom sheet shows
    		- If not cross signed
    			- [red shield] NOT Verified
    			- Shows device name / id 
    			- (action) Manually verify by text
    			- (action) Interactively Verify by Emoji
    			- (action) Sign out of this session
    			- (action) Rename
    		- If cross signed
    			- [green shield]  Verified
    			- Shows device name / id 
    			- (action) Sign out of this session
    			- (action) Rename
    		- If not E2E capable
    			- (No shields / header)
    			- Shows device name / id 
    			- No crytpographic information available
    			- (action) Sign out of this session

 
### And Current session is NOT trusted

	- Current Session:
		- Red Shield, on tap the bottomsheet shows:
			- Verify this login
			- Confirm your identity by verifying this login from one of your other sessions, granting it access to encrypted messages.
			- Shows device name / id 
			- (action) Verify this login 
		- + Shortcut button to Verify this login
		
	- Other Sessions:
		- Shields are all black, on tap the bottomsheet shows:
			- If E2E capable
				- (TBD text?) 
				- (action) Sign out of this session
				- (action) Rename
			- If not E2E capable
				- (No shields / header)
				- Shows device name / id 
				- No crytpographic information available
				- (action) Sign out of this session
 

## If Cross Signing Is not Enabled in the account

    - Current Session:
    		- Green Shield, on tap the bottomsheet shows:
    			- [green shield]  Verified
    			- Shows device name / id 
    			- (action) Rename
    			
    - Other Sessions:
    	- Shields reflect local trust and show same as first case based on local trust

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #1295

## When cross signing is enabled, but current session not verified

image

image

## After verification

image

 - With new options to verify

image

image

 - Manually verify will cross sign and uplaod signature

## Copy updates 
image

image

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/core/ui/list/GenericButtonItem.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @ColorInt
+    var textColor: Int? = null
+
+    @EpoxyAttribute
+    @DrawableRes
+    var iconRes: Int? = null
+
+    override fun bind(holder: Holder) {
+        holder.button.text = text
+        val textColor = textColor ?: ThemeUtils.getColor(holder.view.context, R.attr.colorPrimary)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/core/ui/list/GenericButtonItem.kt:51]: use riotx_text_primary ?

File: vector/src/main/java/im/vector/riotx/features/settings/devices/DeviceVerificationInfoEpoxyController.kt [hunk 2]
```diff
@@ -37,109 +40,324 @@ class DeviceVerificationInfoEpoxyController @Inject constructor(private val stri
...
+            title(info?.displayName ?: "")
+            description("(${info?.deviceId})")
+        }
+
+        genericFooterItem {
+            id("infoCrypto${info?.deviceId}")
+            text(stringProvider.getString(R.string.settings_failed_to_get_crypto_device_info))
+        }
+
+        if (!data.isMine) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/settings/devices/DeviceVerificationInfoEpoxyController.kt:347]: why not calling addGenericDeviceManageActions() ?

File: vector/src/main/java/im/vector/riotx/features/settings/devices/TrustUtils.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+package im.vector.riotx.features.settings.devices
+
+import im.vector.matrix.android.internal.crypto.crosssigning.DeviceTrustLevel
+import im.vector.riotx.R
+
+object TrustUtils {
+
+    fun shieldForTrust(currentDevice: Boolean, trustMSK: Boolean, legacyMode: Boolean, deviceTrustLevel: DeviceTrustLevel?): Int {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/settings/devices/TrustUtils.kt:24]: add @DrawableRes

File: vector/src/main/java/im/vector/riotx/core/ui/list/GenericButtonItem.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/core/ui/list/GenericButtonItem.kt:51]: done

File: vector/src/main/java/im/vector/riotx/features/settings/devices/DeviceVerificationInfoEpoxyController.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/settings/devices/DeviceVerificationInfoEpoxyController.kt:347]: fixed, thx

File: vector/src/main/java/im/vector/riotx/features/settings/devices/TrustUtils.kt (hunk 3 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/settings/devices/TrustUtils.kt:24]: done

**Pull request merged**